File: Front/src/app/models/cuestionario.ts
import { Usuario } from './../../../../Frontend/src/app/models/usuario';
import { Pregunta } from "./pregunta";

export class Cuestionario{
    id?:number;
    nombre:string;
    descripcion:string;
    fechaCreacion?:Date;
    preguntas:Pregunta[];
    usuario?:Usuario;

    constructor(nombre:string,descripcion:string,fechaCreacion:Date,preguntas:Pregunta[],usuario:Usuario) {
        this.nombre =nombre;
        this.descripcion = descripcion;
        this.fechaCreacion = fechaCreacion;
        this.preguntas = preguntas;
        this.usuario = usuario;
    }

}

File: BackEnd/BackEnd/Domain/Model/Cuestionario.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackEnd.Domain.Model
{
    public class Cuestionario
    {
        public int? Id { get; set; }
        [Required]
        [Column(TypeName = "varchar(100)")]
        public string Nombre{ get; set; }
        [Required]
        [Column(TypeName = "varchar(150)")]
        public string Descripcion { get; set;}
        public DateTime? FechaCreacion { get; set;}
        public bool? Activo { get; set; }
        public List<Pregunta> Preguntas { get; set; }
        public int? UsuarioId { get; set; }
        public Usuario? Usuario { get; set; }
    }
}
